bedrock/firefox/views.py

```diff
@@ -242,6 +242,106 @@ def send_to_device_ajax(request):
+        'desktop_beta_latest_version': latest_beta_version_desktop,
+
+        'desktop_developer_platforms': product_desktop.platforms(channel_alpha, True),
+        'desktop_developer_full_builds': product_desktop.get_filtered_full_builds(channel_alpha, latest_developer_version_desktop),
+        'desktop_developer_channel_label': product_desktop.channel_labels.get(channel_alpha, 'Firefox'),
+        'desktop_developer_latest_version': latest_developer_version_desktop,
+
+        'desktop_nightly_platforms': product_desktop.platforms(channel_alpha, True),
+        'desktop_nightly_full_builds': product_desktop.get_filtered_full_builds(channel_nightly, latest_nightly_version_desktop),
+        'desktop_nightly_channel_label': product_desktop.channel_labels.get(channel_nightly, 'Firefox'),
+        'desktop_nightly_latest_version': latest_nightly_version_desktop,
+
+        'desktop_esr_platforms': product_desktop.platforms(channel_esr, True),
+        'desktop_esr_full_builds': product_desktop.get_filtered_full_builds(channel_esr, latest_esr_version_desktop),
+        'desktop_esr_channel_label': product_desktop.channel_labels.get(channel_esr, 'Firefox'),
+        'desktop_esr_latest_version': latest_esr_version_desktop,
+
+        'android_release_platforms': product_android.platforms(channel_release, True),
+        'android_release_full_builds': product_android.get_filtered_full_builds(channel_release, latest_release_version_android),
+        'android_release_channel_label': product_android.channel_labels.get(channel_release, 'Firefox'),
+        'android_release_latest_version': latest_release_version_android,
+
+        'android_beta_platforms': product_android.platforms(channel_beta, True),
+        'android_beta_full_builds': product_android.get_filtered_full_builds(channel_beta, latest_beta_version_android),
+        'android_beta_channel_label': product_android.channel_labels.get(channel_beta, 'Firefox'),
+        'android_beta_latest_version': latest_beta_version_android,
+
+        'android_nightly_platforms': product_android.platforms(channel_nightly, True),
+        'android_nightly_full_builds': product_android.get_filtered_full_builds(channel_nightly, latest_nightly_version_android),
+        'android_nightly_channel_label': product_android.channel_labels.get(channel_nightly, 'Firefox'),
+        'android_nightly_latest_version': latest_nightly_version_android,
+    }
+
+    if latest_esr_next_version_desktop:
+        context['desktop_esr_platforms_next'] = product_desktop.platforms(channel_esr_next, True)
+        context['desktop_esr_full_builds_next'] = product_desktop.get_filtered_full_builds(channel_esr_next, latest_esr_next_version_desktop)
+        context['desktop_esr_channel_label_next'] = product_desktop.channel_labels.get(channel_esr_next, 'Firefox'),
+        context['desktop_esr_next_version'] = latest_esr_next_version_desktop
+
+    return l10n_utils.render(request, 'firefox/all-unified.html', context)
```

Comment:
Wow. It's a bummer all of that had to be written out like that. I'd say we could refactor into some helper functions and/or loops but that would probably be a lot harder to grok for future devs. Good work.